library/src/main/java/com/davidstemmer/screenplay/flow/Screenplay.java

```diff
@@ -1,145 +1,139 @@
 package com.davidstemmer.screenplay.flow;
 
+import android.app.Activity;
 import android.view.View;
 import android.view.ViewGroup;
 
-import com.davidstemmer.screenplay.SceneCut;
 import com.davidstemmer.screenplay.SceneState;
-import com.davidstemmer.screenplay.Stage;
 import com.davidstemmer.screenplay.scene.Scene;
+import com.davidstemmer.screenplay.util.CollectionUtils;
 
 import java.util.ArrayDeque;
+import java.util.Collection;
+import java.util.Deque;
 import java.util.Iterator;
 
-import flow.Backstack;
-import flow.Flow;
-
 
 /**
  * The Screenplay object handles the navigation logic for a Screenplay application.
  */
-public class Screenplay implements Flow.Listener {
+public class Screenplay {
 
     private SceneState screenState = SceneState.NORMAL;
 
-    private final Stage stage;
-
-    private final ArrayDeque<Scene> previousBackstack = new ArrayDeque<>();
+    private final Activity activity;
+    private final ViewGroup container;
 
-    public Screenplay(Stage stage) {
-        this.stage = stage;
+    public Screenplay(Activity activity, ViewGroup container) {
+        this.activity = activity;
+        this.container = container;
     }
 
-    @Override
-    public void go(Backstack backstack, Flow.Direction direction, Flow.Callback callback) {
+    public enum Direction {
```

Comment:
Iv been noticing a lot of hatred towards enums across the android community due to them taking up more memory than an integer. Maybe we can make this an int and use the Android annotation @Intdef to only allow certain integers to be passed in?